odl/solvers/functional/functional.py

```diff
@@ -0,0 +1,1671 @@
+
+        Parameters
+        ----------
+        sigma : positive float, optional
+            Regularization parameter of the proximal operator.
+
+        Returns
+        -------
+        out : `Operator`
+            Domain and range equal to domain of functional.
+        """
+        return (proximal_translation(self._original_func.proximal,
+                                     self._translation))(sigma)
+
+    @property
+    def conjugate_functional(self):
+        """Convex conjugate functional of the translated functional."""
+        return ConvexConjugateTranslation(
+            self._original_func.conjugate_functional,
+            self._translation)
+
+    def derivative(self, point):
+        """Returns the derivative operator, evaluated in the point
+        ``point - translation``.
+
+        Parameters
+        ----------
+        point : `LinearSpaceVector`
+            The point in which the gradient is evaluated.
+
+        Returns
+        -------
+        out : `Operator`
+            The linear operator that maps ``x --> <x, grad_f(p)>``.
+        """
+        return self._original_func.derivative(point - self._translation)
+
+
+class ConvexConjugateTranslation(Functional):
+
+    """ The ``Functional`` representing ``(F( . - translation))^*``.
+
+    This is a functional representing the conjugate functional of the
+    translated function ``F(. - translation)``. It is calculated according to
+    the rule
+
+        ``(F( . - translation))^* (x) = F^*(x) + <translation, x>``,
+
+    where ``translation`` is the translation of the argument.
+
+    Parameters
+    ----------
+    convex_conj_f : `Functional`
+        Function corresponding to ``F^*``.
+
+    translation : `LinearSpaceVector`
+        Element in domain of ``F^*``..
+
+    Notes
+    -----
+    The implementation assumes that the underlying  functional ``F`` is proper,
+    convex, and lower semi-continuous. Otherwise the convex conjugate of the
+    convex conjugate is not necessarily the functional itself.
+
+    For reference on the identity used, see [KP2015]_.
+    """
+
+    def __init__(self, convex_conj_f, translation):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        convex_conj_f : `Functional`
+            Function corresponding to F^*.
+
+        translation : `LinearSpaceVector`
+            Element in domain of ``F^*``.
+        """
+
+        if not isinstance(convex_conj_f, Functional):
+            raise TypeError('input {} is not a Functional instance'
+                            ''.format(convex_conj_f))
+
+        if translation is not None and not isinstance(translation,
+                                                      LinearSpaceVector):
+            raise TypeError(
+                'vector {!r} not None or a LinearSpaceVector instance.'
+                ''.format(translation))
+
+        if translation not in convex_conj_f.domain:
```

Comment:
This crashes for `None`. Take this check and replace the above `and not isinstance(...)` with it.